### PR TITLE
GH-1326 Move Sink code to new Namespace

### PR DESCRIPTION
**Pull request by okotsopoulos**
### Purpose

- https://broadinstitute.atlassian.net/browse/GH-1326

And fleshed out developer debug documentation in response to a roadblock I encountered (thanks @tbl3rd for the linting tip!)

### CODE REVIEW

**Review by ehigham — APPROVED** [review]
Looks good. Thanks. I think the failures are unrelated so happy for you to merge this.

File: api/src/wfl/executor.clj [hunk 1]
```diff
@@ -43,7 +43,7 @@
      in the database schema.
    - This multimethod is type-dispatched on the `:name` association in the
      `request`."
-  (fn [_transaction _id request] (:name request)))
+  (fn [_transaction _workload-id request] (:name request)))
```
Comment (rexwangcc) [inline api/src/wfl/executor.clj:46]: thanks!

File: api/test/wfl/integration/executor_test.clj [hunk 2]
```diff
@@ -83,7 +83,7 @@
          :methodConfigurationVersion -1
          :fromSource                 "importSnapshot"}))))
 
-(deftest test-validate-terra-executor--with-wrong-method-configuration
+(deftest test-validate-terra-executor-with-wrong-method-configuration
```
Comment (rexwangcc) [inline api/test/wfl/integration/executor_test.clj:86]: wow, thanks!

File: docs/md/dev-process.md [hunk 3]
```diff
@@ -283,3 +283,59 @@ It takes a vector of `var`s like this.
 ``` clojure
 (comment (test-vars [#'test-something]))
 ```
+
+### No tests found
```
Comment (rexwangcc) [inline docs/md/dev-process.md:287]: Thank you for adding this!! I ran into this without a clue just yesterday!!

**Review by rexwangcc — APPROVED** [review]
LGTM!! Tests are failing I believe due to [this](https://broadinstitute.slack.com/archives/CD4HBRFMG/p1624548823224100). Feel free to merge.

File: docs/md/dev-process.md (hunk 3 above)
Comment (okotsopoulos) [reply docs/md/dev-process.md:287]: A first time for me as well!  It lulls you into a false sense of security.

**Pull request merged**